forte/processors/base/data_augment_processor.py

```diff
@@ -0,0 +1,92 @@
+# limitations under the License.
+"""
+Processors that augment the data. The processor will call
+replacement ops to generate texts similar to those in the input pack
+and create a new pack with them.
+"""
+from typing import List, Tuple
+from forte.data.ontology.core import Entry
+from forte.processors.base.base_processor import BaseProcessor
+from forte.processors.data_augment.algorithms.text_replacement_op \
+    import TextReplacementOp
+
+__all__ = [
+    "BaseDataAugmentProcessor",
+    "ReplacementDataAugmentProcessor"
+]
+
+
+class BaseDataAugmentProcessor(BaseProcessor):
+    r"""The base class of processors that augment data.
+    This processor instantiates replacement ops where specific
+    data augmentation algorithms are implemented. The replacement ops
+    will run the algorithms and the processor will create Forte
+    data structures based on the augmented inputs.
+    """
+
+
+class ReplacementDataAugmentProcessor(BaseDataAugmentProcessor):
+    r"""
+    Most of the Data Augmentation(DA) methods can be
+    considered as replacement-based methods with different
+    levels: character, word, sentence or document.
+    """
+    def __init__(self):
+        """
+        The replaced_spans records the entries replaced by new texts.
+        It consists of tuples (entry, new text) inserted by :func: replace.
+        The new text will be used for building new data pack.
+        """
+        super().__init__()
+        self.replaced_spans: List[Tuple[Entry, str]] = []
+
+    def replace(self, replacement_op: TextReplacementOp, input: Entry):
+        """
+        This is a wrapper function to call the replacement op. After
+        getting the augmented text, it will register the input & output
+        for later batch process of building the new data pack.
+        """
+        replaced_text: str = replacement_op.replace(input)
+        self.replaced_spans.append((input, replaced_text))
+
+    @classmethod
+    def default_configs(cls):
+        """
+        Returns:
+            A dictionary with the default config for this processor.
+        Following are the keys for this dictionary:
+            - augment_entries: defines the entries the processor
+            will augment. It should be a full path to the entry class.
+            - other_entry_policy: a dict specifying the policies
+            for other entries that is not the "augment_entry".
+            If "auto_align", the span will be automatically modified according
+            its original location. However, some spans might become invalid
+            after the augmentation, for example, the tokens within a replaced
+            sentence may disappear.
+            If "delete", the entry will not be copied to the new data pack. It
+            is the default operation for entries not specified in this dict.
+
+            Example: {
+                "ft.onto.base_ontology.Sentence": "auto_align",
+                "ft.onto.base_ontology.Document": "delete"
```

Comment:
I think users won't need to define all entries, we simply not copy the items that the users don't provide a resolution here.

Reply:
Yes. As in the comment we said, the "delete" is the default operation for entries not specified in the dict.

Reply:
Now this logic is not clear.

First, we discuss that we will not use the word "Delete", because 1) this make users feel there would be data loss 2) this is not the same as our code behavior. What the code is doing is simply "Not Copy".

Second, there is no need to provide a key for "not going to copy" entries. And in the documentation, we shouldn't say the default behavior is "Delete", we should simply say we will copy the entries that you specify here. When would the user ever need to specify "Delete"?

Reply:
That makes sense. In this case, I don't think we need the dict as well. We only need a list, and all the entries in the list should be "auto_align". For entries not on the list, we do not copy them.